chapters/03_queueing_model/main.tex

```diff
@@ -0,0 +1,252 @@
+\ref{fig:general_markov_model}.
+
+\input{chapters/03_queueing_model/img/general_markov_model.tex}
+
+
+In order to consider this model numerically an adjustment needs to be made.
+The problem defined above assumes no upper boundary to the number of individuals
+that can wait for service or for the ones that are blocked in waiting zone 2.
+Therefore, a different state space \( \tilde S \) is constructed where
+\( \tilde S \subseteq S \) and there is a maximum allowed number of individuals
+\(N\) that can be in waiting zone 1 and a maximum allowed number of individuals
+\(M\) that can be blocked in waiting zone 2:
+
+\begin{equation}\label{eq:truncated_state_space}
+    \tilde S = \left\{ (u, v) \in S\;| u \leq M, v\leq N \right\}
+\end{equation}
+
+The adjusted Markov chain model with states \(\tilde S\) can be seen in Figure
+\ref{fig:adjusted_markov_model}.
+
+\input{chapters/03_queueing_model/img/general_adjusted_markov_model.tex}
+
+
+
+\subsubsection{Steady state probability vector \(\pi\)}
+The generator matrix \( Q \) defined in (\ref{eq:markov_transition_rate}) can
+be used to get the probability vector \( \pi \) that contains the steady state
+probabilities of the Markov chain model.
+The vector \( \pi \) is commonly used to study stochastic systems and it's main
+purpose is to keep track of the probability of being at any given state of
+the Markov chain model.
+\(\pi_i\) is the steady state probability of being in state \((u_i, v_i) \in
+\tilde S\) which is the \(i^{\text{th}}\) state of \(\tilde S\) for some
+ordering of \(\tilde S\).
+The term \textbf{steady state} refers to the instance of the vector \( \pi \)
+where the probabilities of being at any state becomes stable over time.
+Thus, by considering the steady state vector \( \pi \) the relationship between
+it and \( Q \) is given by:
+
+\begin{equation}\label{eq:steady_state_from_generator_matrix}
+    \frac{d\pi}{dt} = \pi Q = \vec{0}
+\end{equation}
+
+\subsubsection{Linear algebraic approach}
+The steady state probability vector \( \pi \) can be obtained by solving the
+linear equation:
+
+\begin{equation}\label{eq:numpy_linalg_solve_1}
+    Q^T \pi = \vec{0} \hspace{0.5cm} \text{such that} \hspace{0.5cm} 
+    \sum_{i} \pi_i = 1
+\end{equation}
+
+The two equations can be combined into one by augmenting the matrix \( Q^T \)
+in such a way that it includes the extra constraint \( \sum_i \pi_i = 1 \).
+The new augmented matrix \(\tilde Q\) is defined as \(Q\) with the final
+column replaced with a vector of ones.
+Note that, \(\tilde Q\) needs to be a square matrix in order to solve the
+equation using linear algebra (i.e. the matrix needs to be invertible).
+
+Thus, the steady state probability vector can be calculated by solving the
+linear equation:
+
+\begin{equation}
+    \tilde Q^T \pi = \vec{0}
+\end{equation}
+
+The function \textit{numpy.linalg.solve} from the python library \textit{numpy}
+was used for this calculation \cite{2020NumPy-Array}.
```

Comment:
Avoid talking about implementation function. Discuss the underlying mathematical technique (whatever it is that `np.linalg.solve` does) and mention that it's implemented using `np.linalg.solve`.